sdk/core/azure-core/src/main/java/com/azure/android/core/util/DateTimeRfc1123.java

```diff
@@ -48,6 +55,170 @@ public OffsetDateTime getDateTime() {
         return this.dateTime;
     }
 
+    /**
+     * Parses the RFC1123 format datetime string into OffsetDateTime.
+     *
+     * @param date The datetime string in RFC1123 format
+     * @return The underlying OffsetDateTime.
+     *
+     * @throws DateTimeException If the processing character is not a digit character.
+     * @throws IllegalArgumentException if the given character is not recognized in the pattern of Month. such as 'Jan'.
+     * @throws IndexOutOfBoundsException if the {@code beginIndex} is negative, or beginIndex is larger than length of
+     *   {@code date}.
+     */
+    private static OffsetDateTime parse(final String date) {
+        try {
+            return OffsetDateTime.of(
+                parseInt(date, 12, 16),  // year
+                parseMonth(date, 8), // month
+                parseInt(date, 5, 7),    // dayOfMonth
+                parseInt(date, 17, 19),  // hour
+                parseInt(date, 20, 22),  // minute
+                parseInt(date, 23, 25),  // second
+                0,                    // nanoOfSecond
+                org.threeten.bp.ZoneOffset.UTC);
+        } catch (DateTimeException | IllegalArgumentException | IndexOutOfBoundsException e) {
+            return OffsetDateTime.parse(date, DateTimeFormatter.RFC_1123_DATE_TIME);
+        }
+    }
+
+    /**
+     * Parses the specified substring of datetime to a 'int' value.
+     *
+     * @param date The datetime string in RFC1123 format.
+     * @param beginIndex The beginning index, inclusive.
+     * @param endIndex The ending index, exclusive.
+     * @return The specified substring.
+     *
+     * @throws DateTimeException If the processing character is not digit character.
+     */
+    private static int parseInt(final CharSequence date, final int beginIndex, final int endIndex) {
+        int num = 0;
+        for (int i = beginIndex; i < endIndex; i++) {
+            final char c = date.charAt(i);
+            if (c < '0' || c > '9') {
+                throw new DateTimeException("Invalid date time: " + date);
```

Comment:
We should probably [follow Java](https://github.com/Azure/azure-sdk-for-java/blob/042aa47cf0f30dec6c06d9534d9840fb197a4d06/sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java#L110) here and log these exceptions.
```suggestion
                throw LOGGER.logExceptionAsError(new DateTimeException("Invalid date time: " + date));
```